frontend/lib/algolia/server/findDescendantDevicesWithProducts.ts

```diff
@@ -0,0 +1,39 @@
+import {
+   ALGOLIA_API_KEY,
+   ALGOLIA_APP_ID,
+   ALGOLIA_PRODUCT_INDEX_NAME,
+} from '@config/env';
+import { CLIENT_OPTIONS, escapeFilterValue } from '@helpers/algolia-helpers';
+import { timeAsync } from '@ifixit/stats';
+import { cache } from '@lib/cache';
+import algoliasearch from 'algoliasearch';
+
+export async function findDescendantDevicesWithProducts(device: string) {
+   return cache(
+      `algolia.findDescendantDevicesWithProducts.${device}`,
+      () =>
+         timeAsync('algolia.findDescendantDevicesWithProducts', () =>
+            fetchDevices(device)
+         ),
+      60 * 10
+   );
```

Comment:
It's probably not worth it to apply in-memory cache, since we're caching at most 100 items (across the board) so likely it wouldn't impact much. Beside, once I add the Redis cache this will be already covered by that

Reply:
Good call 👍🏻 https://github.com/iFixit/react-commerce/pull/1252/commits/2b57a1d0859dac53801995c7fe0c4509387ee93f